client/src/components/Workout/NewWorkout/hooks/useNewWorkout.tsx

```diff
@@ -1,54 +1,77 @@
-import { useCallback, useReducer } from "react";
-import { useRecoilState, useRecoilValue } from "recoil";
+import { ChangeEvent, useCallback, useEffect, useReducer, useRef } from "react";
+import { useRecoilState, useResetRecoilState, useSetRecoilState } from "recoil";
 import useCreateWorkout from "../../../../helpers/fetch/workouts/useCreateWorkout";
+import useRouterProps from "../../../../hooks/useRouterProps";
 import { splitNameAndIndex } from "../helpers/field-with-index";
 import { parseNewWorkout } from "../helpers/parse";
-import { newWorkoutState, weightUnitState } from "../state/new-workout-state";
+import { isValidNewWorkout } from "../helpers/validate";
+import {
+	newWorkoutState,
+	openIndexState,
+	weightUnitState,
+} from "../state/new-workout-state";
 import NewExercise from "../sub/NewExercise";
 
 export default function useNewWorkout() {
-	const [newWorkout, setNewWorkout] = useRecoilState(newWorkoutState);
 	const { mutate } = useCreateWorkout();
-	const weightUnit = useRecoilValue(weightUnitState);
+	const { navigate } = useRouterProps();
+	const [weightUnit, setWeightUnit] = useRecoilState(weightUnitState);
+	const [newWorkout, setNewWorkout] = useRecoilState(newWorkoutState);
+	const resetNewWorkout = useResetRecoilState(newWorkoutState);
+	const isValid = isValidNewWorkout(parseNewWorkout(newWorkout, weightUnit));
+	const setOpenIndex = useSetRecoilState(openIndexState);
+	const deleteCount = useRef(0);
+
+	useEffect(() => {
+		return () => resetNewWorkout();
+	}, []);
 
 	/** Reducer to manipulate `elements` state. Note that this has to be defined
 	 * inside this component so that it has access to local state setters. */
-	function fieldsetElementReducer(state: JSX.Element[], action: string): JSX.Element[] {
+	function exerciseElementsReducer(
+		state: JSX.Element[],
+		action: { type: string; index?: number }
```

Comment:
Maybe call the parameter `id` instead of `index`, since it's not an array index anymore.